src/Build/BackEnd/Components/Communications/NodeProviderOutOfProc.cs

```diff
@@ -173,13 +180,20 @@ public void ShutdownConnectedNodes(bool enableReuse)
         /// </summary>
         public void ShutdownAllNodes()
         {
-            // if no BuildParameters were specified for this build,
+            // If no BuildParameters were specified for this build,
             // we must be trying to shut down idle nodes from some
             // other, completed build. If they're still around,
             // they must have been started with node reuse.
             bool nodeReuse = ComponentHost.BuildParameters?.EnableNodeReuse ?? true;
 
-            ShutdownAllNodes(NodeProviderOutOfProc.GetHostHandshake(nodeReuse), NodeProviderOutOfProc.GetClientHandshake(), NodeContextTerminated);
+            // If no BuildParameters were specified for this build,
+            // then they couldn't have been launched with priority since it defaults to off.
+            bool lowPriority = ComponentHost.BuildParameters?.LowPriority ?? false;
+
+            ShutdownAllNodes(
+                NodeProviderOutOfProc.GetHostHandshake(nodeReuse, lowPriority),
```

Comment:
I'm worried about only shutting down matching-priority nodes. That would make a scenario like `dotnet build -low && dotnet build-server shutdown` fail, because the second call to `ShutdownAllNodes` would not be at low priority.

We could call twice with low and not-low. But that would preclude the handshake reading priority directly.

We're also a bit concerned about the effect of such a change on VS shutdown time, though we don't have clear numbers to indicate that it would be a problem. That could probably be solved by moving the try-it-both-ways approach inside the part of the loop where we attempt to connect to enumerated processes.

Reply:
I went with the path of trying to connect with and without the priority in the handshake.